azurelinuxagent/ga/exthandlers.py

```diff
@@ -1215,6 +1217,11 @@ def launch_command(self, cmd, timeout=300, extension_error_code=ExtensionErrorCo
                             ExtCommandEnvVariable.ExtensionVersion: str(self.ext_handler.properties.version),
                             ExtCommandEnvVariable.ExtensionSeqNumber: str(self.get_seq_no())})
 
+                python_symlink_path = get_python_symlink_path_if_ubuntu_20_04_image()
+                if python_symlink_path is not None:
```

Comment:
Should we be checking that it's a symlink here like you are in `update.py` before appending it to the PATH?

Reply:
I'm checking there only to verify if it exists or not. I dont want to add that check here because I dont want to manipulate with the filesystem in this method (mainly coz its called so frequently). This check is mainly to verify if its a ub20 image or not because the function `get_python_symlink_path_if_ubuntu_20_04_image` would return not None only if its a ub20 image.